Sources/Receipt/ListReceiptViewPresenter.swift

```diff
@@ -110,13 +168,30 @@ final class ListReceiptViewPresenter {
                                                 .firstDayOfMonth()
         })
 
-        for section in groupedSections {
+        let sortedGroupedSections = groupedSections
+            .sorted(by: prepaidCardToken == nil ? { $0.key > $1.key } : { $0.key < $1.key })
+
+        for section in sortedGroupedSections {
             if let sectionIndex = sectionData.firstIndex(where: { $0.key == section.key }) {
                 sectionData[sectionIndex].value.append(contentsOf: section.value)
             } else {
                 sectionData.append(section)
             }
         }
-        sectionData = sectionData.sorted(by: { $0.key > $1.key })
+    }
+
+    private func filterDuplicatedReceipts(receipts: [HyperwalletReceipt]) -> [HyperwalletReceipt]? {
```

Comment:
I'm not sure that we should filter data from the API server on the client side. It is a server-side responsibility. I think we should show all the data we received.

Reply:
Server side should implement pagination logic properly, but for now it's not ready so this is just a workaround.